flink-cdc-runtime/src/main/java/com/ververica/cdc/runtime/parser/FlinkSqlParser.java

```diff
@@ -0,0 +1,203 @@
+        tableConfig.setSqlDialect(SqlDialect.DEFAULT);
+        CalciteConfig calciteConfig = getCalciteConfig(tableConfig);
+        SqlParser.Config sqlParserConfig =
+                JavaScalaConversionUtil.<SqlParser.Config>toJava(calciteConfig.getSqlParserConfig())
+                        .orElseGet(
+                                // we use Java lex because back ticks are easier than double quotes
+                                // in
+                                // programming and cases are preserved
+                                () -> {
+                                    SqlConformance conformance = FlinkSqlConformance.DEFAULT;
+                                    return SqlParser.config()
+                                            .withParserFactory(
+                                                    FlinkSqlParserFactories.create(conformance))
+                                            .withConformance(conformance)
+                                            .withLex(Lex.JAVA)
+                                            .withIdentifierMaxLength(256);
+                                });
+        return new CalciteParser(sqlParserConfig);
+    }
+
+    public static SqlSelect parseSelect(String statement) {
+        SqlNode sqlNode = calciteParser.parse(statement);
+        if (sqlNode instanceof SqlSelect) {
+            return (SqlSelect) sqlNode;
+        } else {
+            throw new ParseException("Only select statements can be parsed.");
+        }
+    }
+
+    // Parse all columns
+    public static List<ColumnTransform> generateColumnTransforms(String projection) {
+        List<ColumnTransform> columnTransformList = new ArrayList<>();
+        if (StringUtils.isNullOrWhitespaceOnly(projection)) {
+            return columnTransformList;
+        }
+        SqlSelect sqlSelect = FlinkSqlParser.parseProjection(projection);
+        if (sqlSelect.getSelectList().isEmpty()) {
+            return columnTransformList;
+        }
+        for (SqlNode sqlNode : sqlSelect.getSelectList()) {
+            if (sqlNode instanceof SqlBasicCall) {
+                SqlBasicCall sqlBasicCall = (SqlBasicCall) sqlNode;
+                if (SqlKind.AS.equals(sqlBasicCall.getOperator().kind)) {
+                    SqlBasicCall transform = null;
+                    String columnName = null;
+                    List<SqlNode> operandList = sqlBasicCall.getOperandList();
+                    for (SqlNode operand : operandList) {
+                        if (operand instanceof SqlBasicCall) {
+                            transform = (SqlBasicCall) operand;
+                        } else if (operand instanceof SqlIdentifier) {
+                            columnName = ((SqlIdentifier) operand).getSimple();
+                        }
+                    }
+                    columnTransformList.add(
+                            ColumnTransform.of(columnName, DataTypes.STRING(), transform));
```

Comment:
Set all column to string type will cause inaccuracies. We need to add type inference, or simply extract one data type from relevant columns.